frame/sudo/src/lib.rs

```diff
@@ -134,6 +134,30 @@ decl_module! {
 			Self::deposit_event(RawEvent::Sudid(res.map(|_| ()).map_err(|e| e.error)));
 		}
 
+		/// Authenticates the sudo key and dispatches a function call with `Root` origin.
+		/// This function does not check the weight of the call, and instead allows the
+		/// Sudo user to specify the weight of the call.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - The weight of this call is defined by the caller.
+		/// # </weight>
+		#[weight = FunctionOf(
+			|(_, &weight): (&Box<<T as Trait>::Call>,&Weight,)| weight,
+			|(call, _): (&Box<<T as Trait>::Call>,&Weight,)| call.get_dispatch_info().class,
+			Pays::Yes,
+		)]
+		fn sudo_unchecked_weight(origin, call: Box<<T as Trait>::Call>, _weight: Weight) {
```

Comment:
Why would anyone use this function and not the one that uses the proper weight? I feel like this question should be answered in the comments above.

Reply:
I thought the comments did answer this?

A very specific example is doing a runtime upgrade. Since I changed the weight of `set_code` to `T::MaximumBlockWeight`, it is no longer callable by regular sudo because a transaction is never allowed to go over the block weight limit.

Reply:
The comment only answers the "what" but not the "why". The why would be as you described:

> This can be used to dispatch calls that are to weighty to be dispatched as a regular transaction. For example `set_code` which uses the full block.

Should this function stay even after making sudo use the scheduler?

Reply:
You can not prevent anything and in the end if you need to fix something, you don't want to mess around with weights. Having one block that may takes more than maximum weight is okay, as this can not be triggered by every user, only by the super user.

Reply:
I understand that we want some kind of emergency sudo which just works. However that does not answer my question: Should this functionality stay even after making sudo use the scheduler? I am asking because it seems to be accomplishing a similar thing (allow overweight transactions).

Reply:
The scheduler will apply it from the next block on and there could may be stuff that is applied earlier. So, I don't think that both accomplish the same functionality.

Reply:
I think it is for handyness, for developers, when you want to use sudo without crafting your call to use scheduler.

Reply:
Also when you dont include scheduler in your runtime for whatever reason.

Reply:
> Why would anyone use this function and not the one that uses the proper weight?

I am not really up to date with the weight system, so it is probably related, but I was asking myself the opposite question:
Why would anyone use `sudo` when they could call `sudo_unchecked_weight` with a weight of 0?